src/Page/PageServiceManager.php

```diff
@@ -109,11 +110,16 @@ public function execute(PageInterface $page, Request $request, array $parameters
      */
     private function createResponse(PageInterface $page)
     {
-        if ($page->getTarget()) {
-            $page->addHeader('Location', $this->router->generate($page->getTarget()));
```

Comment:
@eerison can you try if this code is used on any of your projects? `$page->getTarget()` returns a PageInterface, but the router generate expects a string...

Reply:
Hey @jordisala1991 I checked and this target is defined only in snapshot entity: https://github.com/sonata-project/SonataPageBundle/blob/4.x/src/Resources/config/doctrine/BaseSnapshot.orm.xml#L13

But I'm my project all data have this field null, then it explain why this never throw an exception because it never going into the "if" condition, and it's used in every page.

should I create an issue to check this more in deep and see if we can remove this field?

Reply:
if it's not used then no make sense to have this `createResponse` method

Reply:
if you want I can care about this :)

Reply:
I just asked for some other guys check in their projects if there is some id in `target_id`, But I checked in other project used for several years and there isn't any row with `target_id`

Reply:
#1481 OK I created this issue to investigate a bit more about this `target_id`, when the 4.x branch is able to use sonata 4, I'll stop to work on that ticket and back for the tickets that are necessary for #1416

Reply:
Hey @jordisala1991 this method can be remove on `4.x` ;) 

https://github.com/sonata-project/SonataPageBundle/pull/1482/files#diff-46f2a9ad831e7f18ef69582b9678dd4b7b7574820ef3fc78481de7ac7884ca8bR116

Reply:
and just the 

```
new Response('', 200, $page->getHeaders() ?: []);
```

directly instead of `createResponse`

Reply:
Fixed.